cpp/include/cudf/hashing.hpp

```diff
@@ -125,6 +126,24 @@ std::unique_ptr<column> md5(
   rmm::cuda_stream_view stream        = cudf::get_default_stream(),
   rmm::mr::device_memory_resource* mr = rmm::mr::get_current_device_resource());
 
+/**
+ * @brief Computes the hash value of each row in the input set of columns
+ *
+ * This function takes a 64-bit seed value and returns a column of type UINT64.
+ *
+ * @param input The table of columns to hash
+ * @param seed Optional seed value to use for the hash function
+ * @param stream CUDA stream used for device memory operations and kernel launches
+ * @param mr Device memory resource used to allocate the returned column's device memory
+ *
+ * @returns A column where each row is the hash of a row from the input
+ */
+std::unique_ptr<column> xxhash64(
```

Comment:
I based this code on https://github.com/NVIDIA/cuCollections/blob/dev/include/cuco/detail/hash_functions/xxhash.cuh
in case that matters. I really could not follow the source code for the link you provided.

Reply:
@sleeepyjack Do you know which algorithm the cuCo implementation is based on?

Reply:
The code matches more to the cuCollections implementation and I foresee any updates would come from there as well. 
Regardless, I have no preference on the names we use here.

Reply:
Sounds fine to me.

Reply:
> @sleeepyjack Do you know which algorithm the cuCo implementation is based on?

Oh wow, I found this comment by chance. Somehow I don't get notifications from the RAPIDS org since we moved to GH enterprise -.-

> I'm wondering why we chose to implement XXHash_64 rather than XXH3, if that's what is being used in cuCo.

My understanding is that XXH3 is the SSE2 version of XXH64. Since we don't use explicit vectorization, I think XXHash_64 or XXH64 is the right term.